### ISSUE TITLE
Run commands on create/provision on host

### ISSUE THREAD

**Issue by garethr** [docker/machine#3188]
I would like to be able to run arbitrary commands when a machine is created by docker-machine. My usecase for this is setting up the underlying machine (in my case with Puppet, but it could be any tool of choice including bash). For instance you may want to:

* Create a local user
* Configure auditd
* Add specific firewall rules

At the moment this is possible (assuming these can run after Docker is installed) by:

* Creating custom images to be booted
* Running the command via `docker-machine ssh` after boot

Ideally I'd rather be able to do with this without a custom image (more portable, especially for examples and documentation) and without a second command (nicer user experience, less space for error)

I would see this as a new flag on create which applies to all drivers and takes either:

* an argument of a script which is uploaded and executed post provision
* an argument of a command which is executed post provision

```
docker-machine create --driver virtualbox default --exec /usr/local/bin/existing-command
docker-machine create --driver virtualbox default --exec-file upload-and-run-this-file.sh
```

I'd likely be interested in working on this but wanted to check whether this is something of interest first, and whether the suggested approach works.

**Comment by cadavre** [comment]
Pre-provision and post-provision scripts could be useful. In some cases we can now specify userdata scripts, but it is called on machine provision when Docker is not yet installed.

Use case: create shared volume that can be mounted after machine was successfully provisioned.

**Comment by scizeron** [comment]
Yes, it would be interesting. Behind a corporate proxy, I need to add some ssl certificates, update ca-certificates.crt and restart docker to apply. So, definitively a good thing because I need to add several commands with docker-machine ssh.
